Assets/Countly/Plugins/CountlySDK/Interfaces/IViewModule.cs

```diff
@@ -1,26 +1,126 @@
 using System;
-using System.Collections.Generic;
-using System.Threading.Tasks;
-
-// interface for SDK users
-public interface IViewModule
-{
-    /// <summary>
-    /// Start tracking a view
-    /// </summary>
-    /// <param name="name">name of the view</param>
-    /// <returns></returns>
-    [Obsolete("RecordOpenViewAsync(string name, IDictionary<string, object> segmentation = null) is deprecated, this is going to be removed in the future.")]
-    Task RecordOpenViewAsync(string name, IDictionary<string, object> segmentation = null);
-
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+// interface for SDK users
+public interface IViewModule
+{
+    /// <summary>
+    /// Starts a view which would not close automatically
+    /// </summary>
+    /// <param name="viewName"></param>
+    /// <returns></returns>
+    string StartView(string viewName);
+
+    /// <summary>
+    /// Starts a view which would not close automatically
+    /// </summary>
+    /// <param name="viewName">name of the view</param>
+    /// <param name="viewSegmentation">segmentation that will be added to the view, set 'null' if none should be added</param>
+    /// <returns></returns>
+    string StartView(string viewName, Dictionary<string, object> viewSegmentation);
+
+    /// <summary>
+    /// Starts a view which would be closed automaticly
+    /// </summary>
+    /// <param name="viewName"></param>
+    /// <returns></returns>
+    string StartAutoStoppedView(string viewName);
+
+    /// <summary>
+    /// Starts a view which would be closed automaticly
+    /// </summary>
+    /// <param name="viewName"></param>
+    /// <param name="viewSegmentation"></param>
+    /// <returns></returns>
+    string StartAutoStoppedView(string viewName, Dictionary<string, object> viewSegmentation);
+
+    /// <summary>
+    /// Stops a view with the given name if it was open
+    /// </summary>
+    /// <param name="viewName">name of the view</param>
+    void StopViewWithName(string viewName);
+
+    /// <summary>
+    /// Stops a view with the given name if it was open
+    /// </summary>
+    /// <param name="viewName">name of the view</param>
+    /// <param name="viewSegmentation">view segmentation</param>
+    void StopViewWithName(string viewName, Dictionary<string, object> viewSegmentation);
+
+    /// <summary>
+    /// Stops a view with the given ID if it was open
+    /// </summary>
+    /// <param name="viewID">ID of the view</param>
+    void StopViewWithID(string viewID);
```

Comment:
why we are not using c# function overloading? This design selected because to imply with other SDKs?